### PR TITLE
Added support for Mention, Date and Equation in RichText object

### PR DESCRIPTION
**Pull request by sawantshivaji1997**
RichText object can have other object types too like mention a page or a database or user or a date.
It can also have a inline equation.
Ref: https://developers.notion.com/reference/rich-text 

Sample json response for richtext having all the different objects mentioned in above link
```
{
    "object": "block",
    "id": "6c06b362-0000-4a9f-b70d-9181d1751957",
    "parent": {
        "type": "page_id",
        "page_id": "e50c7b3a-0000-4b26-a6f9-6dfef9f74148"
    },
    "created_time": "2022-08-27T07:16:00.000Z",
    "last_edited_time": "2022-08-27T10:42:00.000Z",
    "created_by": {
        "object": "user",
        "id": "00000000-545d-43c2-924e-f18a06fb93b3"
    },
    "last_edited_by": {
        "object": "user",
        "id": "00000000-545d-43c2-924e-f18a06fb93b3"
    },
    "has_children": false,
    "archived": false,
    "type": "paragraph",
    "paragraph": {
        "rich_text": [
            {
                "type": "mention",
                "mention": {
                    "type": "database",
                    "database": {
                        "id": "00000000-545d-43c2-924e-f18a06fb93b3"
                    }
                },
                "annotations": {
                    "bold": false,
                    "italic": false,
                    "strikethrough": false,
                    "underline": false,
                    "code": false,
                    "color": "default"
                },
                "plain_text": "Test DB",
                "href": "https://www.notion.so/00000000510a4756b113cc28c7a30fd7"
            },
            {
                "type": "text",
                "text": {
                    "content": " ",
                    "link": null
                },
                "annotations": {
                    "bold": false,
                    "italic": false,
                    "strikethrough": false,
                    "underline": false,
                    "code": false,
                    "color": "default"
                },
                "plain_text": " ",
                "href": null
            },
            {
                "type": "mention",
                "mention": {
                    "type": "page",
                    "page": {
                        "id": "00000000-545d-43c2-924e-f18a06fb93b3"
                    }
                },
                "annotations": {
                    "bold": false,
                    "italic": false,
                    "strikethrough": false,
                    "underline": false,
                    "code": false,
                    "color": "default"
                },
                "plain_text": "Engineering Guidelines",
                "href": "https://www.notion.so/00000000df684670858ac632ebc2dd02"
            },
            {
                "type": "text",
                "text": {
                    "content": " ",
                    "link": null
                },
                "annotations": {
                    "bold": false,
                    "italic": false,
                    "strikethrough": false,
                    "underline": false,
                    "code": false,
                    "color": "default"
                },
                "plain_text": " ",
                "href": null
            },
            {
                "type": "mention",
                "mention": {
                    "type": "user",
                    "user": {
                        "object": "user",
                        "id": "00000000-545d-43c2-924e-f18a06fb93b3",
                        "name": "Shivaji Sawant",
                        "avatar_url": null,
                        "type": "person",
                        "person": {
                            "email": "sawantshivaji1997@gmail.com"
                        }
                    }
                },
                "annotations": {
                    "bold": false,
                    "italic": false,
                    "strikethrough": false,
                    "underline": false,
                    "code": false,
                    "color": "default"
                },
                "plain_text": "@Shivaji Sawant",
                "href": null
            },
            {
                "type": "text",
                "text": {
                    "content": " ",
                    "link": null
                },
                "annotations": {
                    "bold": false,
                    "italic": false,
                    "strikethrough": false,
                    "underline": false,
                    "code": false,
                    "color": "default"
                },
                "plain_text": " ",
                "href": null
            },
            {
                "type": "mention",
                "mention": {
                    "type": "date",
                    "date": {
                        "start": "2022-08-27T03:00:00.000-07:00",
                        "end": "2022-08-27T03:00:00.000-07:00",
                        "time_zone": null
                    }
                },
                "annotations": {
                    "bold": false,
                    "italic": false,
                    "strikethrough": false,
                    "underline": false,
                    "code": false,
                    "color": "default"
                },
                "plain_text": "2022-08-27T03:00:00.000-07:00 → 2022-08-27T03:00:00.000-07:00",
                "href": null
            },
            {
                "type": "text",
                "text": {
                    "content": " ",
                    "link": null
                },
                "annotations": {
                    "bold": false,
                    "italic": false,
                    "strikethrough": false,
                    "underline": false,
                    "code": false,
                    "color": "default"
                },
                "plain_text": " ",
                "href": null
            },
            {
                "type": "equation",
                "equation": {
                    "expression": "Esd a"
                },
                "annotations": {
                    "bold": false,
                    "italic": false,
                    "strikethrough": false,
                    "underline": false,
                    "code": false,
                    "color": "default"
                },
                "plain_text": "Esd a",
                "href": null
            },
            {
                "type": "text",
                "text": {
                    "content": " ",
                    "link": null
                },
                "annotations": {
                    "bold": false,
                    "italic": false,
                    "strikethrough": false,
                    "underline": false,
                    "code": false,
                    "color": "default"
                },
                "plain_text": " ",
                "href": null
            }
        ],
        "color": "default"
    }
}
```

### CODE REVIEW

File: object.go [hunk 1]
```diff
@@ -34,9 +34,27 @@ func (c Color) MarshalText() ([]byte, error) {
...
+type DatabaseMention struct {
+	ID ObjectID `json:"id"`
+}
+
+type PageMention struct {
+	ID ObjectID `json:"id"`
+}
+
+type Mention struct {
+	Type            ObjectType       `json:"type,omitempty"`
```
Comment (jomei) [inline object.go:46]: This is not an `ObjectType`. I think, it'd be better to make a new type like `MentionType` and add all possible variations into `const.go`

File: object.go [hunk 2]
```diff
@@ -34,9 +34,27 @@ func (c Color) MarshalText() ([]byte, error) {
...
+	DatabaseMention *DatabaseMention `json:"database,omitempty"`
```
Comment (jomei) [inline object.go:47]: Let's keep fields name consistent. So, `DatabaseMention` -> `Database` and `PageMention` -> `Page`

File: object.go [hunk 3]
```diff
@@ -34,9 +34,27 @@ func (c Color) MarshalText() ([]byte, error) {
...
+	PageMention     *PageMention     `json:"page,omitempty"`
+	User            *User            `json:"user,omitempty"`
+	Date            *DateObject      `json:"date,omitempty"`
+}
```
Comment (jomei) [inline object.go:51]: What about [template mention](https://developers.notion.com/reference/rich-text#template-mentions)?
Is it possible to add it also in this PR?

File: object.go [hunk 4]
```diff
@@ -34,9 +34,27 @@ func (c Color) MarshalText() ([]byte, error) {
...
+
 type RichText struct {
 	Type        ObjectType   `json:"type,omitempty"`
-	Text        Text         `json:"text"`
+	Text        *Text        `json:"text,omitempty"`
+	Mention     *Mention     `json:"mention,omitempty"`
+	Equation    *Equation    `json:"equation,omitempty"`
```
Comment (jomei) [inline object.go:57]: And what about [Link preview](https://developers.notion.com/reference/rich-text#link-preview-mentions) mentions?
May be we can add it here also?

**Review by jomei — CHANGES_REQUESTED** [review]
Thanks for your amazing work!

A few suggestions below

**Comment by sawantshivaji1997** [comment]
Thanks for the review. I will work on them.

File: object.go (hunk 1 above)
Comment (sawantshivaji1997) [reply object.go]: Done

File: object.go (hunk 2 above)
Comment (sawantshivaji1997) [reply object.go]: Done

File: object.go (hunk 3 above)
Comment (sawantshivaji1997) [reply object.go:70]: Done

File: object.go (hunk 4 above)
Comment (sawantshivaji1997) [reply object.go:76]: I tried but I was not able to create link preview mentions from Notion. Also, the documentation doesn't provide much information about Link preview schema and how the link preview JSON structure look like.

**Review by jomei — APPROVED** [review]

**Pull request merged**